homeassistant/components/media_player/sonos.py

```diff
@@ -138,9 +138,14 @@ def update(self):
         """Retrieve latest state."""
         self._name = self._player.get_speaker_info()['zone_name'].replace(
             ' (R)', '').replace(' (L)', '')
-        self._status = self._player.get_current_transport_info().get(
-            'current_transport_state')
-        self._trackinfo = self._player.get_current_track_info()
+
+        if self._is_reachable:
+            self._status = self._player.get_current_transport_info().get(
```

Comment:
Wouldn't this call raise an error if we can't connect ? So why not just wrap this block in a `try … except` ?

Also, instead of setting status to unknown and track info empty, consider also exposing the property [`available`](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/helpers/entity.py#L109-L112).

I also see that we assign `self._status` to a SoCo return value. Does that value map to our available states?